spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -956,6 +956,35 @@ for (Pageable p = Pageable.ofSize(25).sortBy(Sort.desc("yearBorn"), Sort.asc("na
 }
 ----
 
+=== Precedence of Repository Methods in Jakarta Data
+
+In Jakarta Data, repository methods define how data is accessed and manipulated. Understanding the precedence of these methods is essential for controlling query execution and customizing data access in your Java applications. It explores the rules governing the precedence of repository methods in Jakarta Data, clarifying how queries are determined and executed.
+
+When working with Jakarta Data repositories, you'll encounter different methods, each with its precedence rules. This guide will help you navigate the hierarchy of these rules, ensuring that your repository methods behave as expected and align with your data access requirements. Whether you're using default method implementations, resource accessor methods, custom queries, or query derivation based on method names and parameters, this section will clarify the order of precedence for each scenario.
+
+.Precedence Rules for Repository Methods
+[cols="2,1"]
+|===
+| Precedence Rule | Description
+
+| If the method is a default method, the provided implementation takes precedence.
+| Default methods provide a base implementation that can be overridden in derived interfaces.
+
+| If the method has no parameters and returns one of (EntityManager/DataSource/Connection) as defined in the spec, then it is a resource accessor method.
+| Resource accessor methods allow direct access to underlying data source resources.
+
+| Otherwise, if the method is annotated with @Query, use the query from the annotation.
+| Annotated methods provide a custom query defined in the @Query annotation.
+
+| Otherwise, if the method name contains the By keyword, determine the query according to the BNF for Query by Method Name.
+| Query by method name allows dynamic query generation based on method names and parameters.
+
+| Otherwise, process it as Query by Parameters, determining the query from the supplied parameters and the insert/update/delete/find/... prefix.
```

Comment:
```suggestion
| Otherwise, process it as Query by Parameters, determining the query from the supplied parameters and the `insert`, `update`, `delete`, `find`, `count`, or `exists` prefix.
```